docs/reference/runtime/runtime-class.md

```diff
@@ -77,7 +77,77 @@ runtime.registerRoutes([
 ]);
 ```
 
-[!ref icon="gear" text="Setup the host application to accept hoisted routes"](/reference/routing/useHoistedRoutes.md)
+[!ref text="Setup the host application to accept hoisted routes"](/reference/routing/useHoistedRoutes.md)
+
+### Register routes under a specific nested layout route
+
+React router [nested routes](https://reactrouter.com/en/main/start/tutorial#nested-routes) enable applications to render nested layouts at various points within the router tree. This is quite helpful for federated applications as it enables composable and decoupled UI.
+
+To fully harness the power of nested routes, the `registerRoutes` function allows a route to be registered **under any** previously registered **nested layout route**, even if that route was registered by another module.
+
+When registering a new route with the `registerRoutes` function, to render the route under a specific nested layout route, specify a `layoutPath` property that matches the nested layout route's `path` property. The only requirement is that the **nested layout route** must has been **registered** to `@squide` **before** the **new child route**.
+
+```tsx !#10
+import { lazy } from "react";
+
+const Page = lazy(() => import("./Page.tsx"));
+
+runtime.registerRoutes([
+    {
+        path: "/layout/page-1",
+        element: <Page />
+    }
+], { layoutPath: "/layout" }); // Register the page under the "/layout" nested layout.
```

Comment:
I liked the idea of having a specific method to register nested layouts, what made you change your mind?

Reply:
Yes it was too hard to come up with the right name and it would have been harder to document. In the end, it seemed fine like that, it's a single function to remember.